### PR TITLE
fix(es/resolver): Handle `TsInterfaceDecl` and `UsingDecl` correctly

### PR DESCRIPTION
**Pull request by l4yton**
Related issue:
- Closes: https://github.com/swc-project/swc/issues/7967

### CODE REVIEW

**Comment by l4yton** [comment]
This should fix the issue, but I'm not sure how to properly add tests for this.

**Comment by kdy1** [comment]
First of all, thank you for the contribution!

For testing, you can add a file named `input.js` to somewhere in https://github.com/swc-project/swc/tree/35fb6528f54b11192dd967297294cdc1852aa57b/crates/swc_ecma_transforms_base/tests/ts-resolver and run `UPDATE=1 cargo test` from your terminal.

Note that you may need to `touch` https://github.com/swc-project/swc/blob/35fb6528f54b11192dd967297294cdc1852aa57b/crates/swc_ecma_transforms_base/tests/ts_resolver.rs

**Comment by kdy1** [comment]
Also, can you sign the CLA?

**Comment by l4yton** [comment]
Sure, will sign the CLA. Just noticed that this fix doesn't work if typescript is `true` when using `resolver(Mark::new(), Mark::new(), true)`, though it does if it's `false`...

**Comment by kdy1** [comment]
I requested review from other team members because I'm not sure about the changes

**Comment by Austaras** [comment]
This looks fine. What concerns you?

File: crates/swc_ecma_transforms_base/src/resolver/mod.rs [hunk 1]
```diff
@@ -344,35 +344,17 @@ impl<'a> Resolver<'a> {
...
-            return;
+        } else {
+            self.current
+                .declared_symbols
+                .insert(ident.sym.clone(), kind);
         }
 
         let mark = self.current.mark;
 
-        self.current
```
Comment (kdy1) [inline crates/swc_ecma_transforms_base/src/resolver/mod.rs:363]: Removal of this logic seems dangerous to me at that time, but now I'm not sure why I feel it dangerous.

**Review by kdy1 — APPROVED** [review]
Thank you!

---


swc-bump:
 - swc_ecma_transforms_base

**Pull request merged**